Examples/ObjectiveC/UserTrackingModesExample.m

```diff
@@ -3,23 +3,152 @@
+        arrow.shouldRasterize = YES;
+        arrow.rasterizationScale = [[UIScreen mainScreen] scale];
+        arrow.drawsAsynchronously = YES;
+        
+        self.arrow = arrow;
+        
+        [self updateArrowStrokeColor:UIColor.whiteColor.CGColor
+             fillColor:UIColor.clearColor.CGColor
+               rotation:0];
+        
+        [self.layer addSublayer:self.arrow];
+    }
+    
+    return self;
+}
+
+- (CGPathRef) arrowPath {
+    
+    CGFloat max = self.buttonSize / 2;
+    
+    UIBezierPath *bezierPath = [[UIBezierPath alloc] init];
+    
+    [bezierPath moveToPoint:CGPointMake(max * 0.5, 0)];
+    
+    [bezierPath addLineToPoint:CGPointMake(max * 0.1, max)];
+    [bezierPath addLineToPoint:CGPointMake(max * 0.5, max * 0.65)];
+    [bezierPath addLineToPoint:CGPointMake(max * 0.9, max)];
+    [bezierPath addLineToPoint:CGPointMake(max * 0.5, 0)];
+    [bezierPath closePath];
+    
+    return bezierPath.CGPath;
+}
+
+-(void) updateArrowStrokeColor:(CGColorRef)strokeColor fillColor:(CGColorRef) fillColor rotation:(CGFloat) rotation {
+    [self.arrow setFillColor:fillColor];
+    [self.arrow setStrokeColor:strokeColor];
+    [self.arrow setAffineTransform:CGAffineTransformMakeRotation(rotation)];
+    
+    [self layoutIfNeeded];
+}
+
+@end
+
+#pragma mark - ViewController
+
+@interface UserTrackingModesExample () <MGLMapViewDelegate>
+
+@property (nonatomic) MGLMapView *mapView;
+@property (nonatomic) UserLocationButton IBOutlet *button;
+
+@end
+
 @implementation UserTrackingModesExample
 
 - (void)viewDidLoad {
     [super viewDidLoad];
 
-    MGLMapView *mapView = [[MGLMapView alloc] initWithFrame:self.view.bounds styleURL:[MGLStyle darkStyleURL]];
-    mapView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
+    self.mapView = [[MGLMapView alloc] initWithFrame:self.view.bounds styleURL:[MGLStyle darkStyleURL]];
+    self.mapView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
+    self.mapView.delegate = self;
 
-    mapView.userTrackingMode = MGLUserTrackingModeFollowWithHeading;
+    // The user location annotation takes its color from the map view's tint color.
+    self.mapView.tintColor = [UIColor redColor];
+    self.mapView.attributionButton.tintColor = [UIColor lightGrayColor];
 
-    // The user location annotation takes its color from the map view tint.
-    mapView.tintColor = [UIColor redColor];
+    [self.view addSubview:self.mapView];
+    
+    [self setupLocationButton];
+}
+
+-(void)locationButtonTapped:(UserLocationButton *)sender {
+    MGLUserTrackingMode mode;
 
-    // You can set the attribution button tint separately.
-    mapView.attributionButton.tintColor = [UIColor lightGrayColor];
+    switch (self.mapView.userTrackingMode) {
+        case MGLUserTrackingModeNone:
+            mode = MGLUserTrackingModeFollow;
+            [sender updateArrowStrokeColor:self.mapView.tintColor.CGColor
```

Comment:
The instinct to consolidate this here is good, but ultimately the appearance of the button should be the responsibility of `UserLocationButton`, not its view controller.